src/graph_world/nodeclassification/benchmarker.py

```diff
@@ -47,9 +47,13 @@ def __init__(self, generator_config, model_class, benchmark_params, h_params):
     self._val_mask = None
     self._test_mask = None
 
-  def AdjustParams(self, generator_config):
-    if 'num_clusters' in generator_config and self._h_params is not None:
-      self._h_params['out_channels'] = generator_config['num_clusters']
+  def AdjustParams(self, generator_config, torch_data):
+    if self._h_params is not None:
+      # Adjusting num_clusters to correct out channels and update config in LFR graphs
+      if generator_config['generator_name']=='LFR':
+        generator_config['num_clusters'] = len(set(torch_data.y.numpy())) 
+      if 'num_clusters' in generator_config:
```

Comment:
This is the node classification benchmarker file, so if I'm not mistaken, `'num_clusters'` should always be a key in the generator config. This check is probably not needed. (Also, if it was needed, it would miss the current L54)